### PR TITLE
user balloon_eval for hover information

### PR DESCRIPTION
**Pull request by prabirshrestha**
based of @mattn changes in https://github.com/mattn/vim-lsp-1/tree/balloon-eval

### CODE REVIEW

File: plugin/lsp.vim [hunk 1]
```diff
@@ -4,6 +4,7 @@ endif
 let g:lsp_loaded = 1
 
 let g:lsp_auto_enable = get(g:, 'lsp_auto_enable', 1)
+let g:lsp_hover_balloon_eval = get(g:, 'lsp_hover_balloon_eval', has('balloon_eval'))
```
Comment (prabirshrestha) [inline plugin/lsp.vim]: need to support this scenario. `let g:lsp_hover_ui = 'echo,preview,balloon'`

**Comment by mattn** [comment]
Please wait merge until fixed a bug that vim call API for balloon with free-ed pointer.

**Comment by prabirshrestha** [comment]
Can you add a reference/link to the vim bug/discussion here for future reference.

**Comment by mattn** [comment]
I confirmed two problems about tooltip.

1. https://github.com/vim-jp/issues/issues/1092
2. Broken text

About No2, I don't confirm how to reproduce.

**Comment by mattn** [comment]
https://github.com/vim/vim/pull/3708 was merged.

**Comment by mattn** [comment]
Use should want to enable balloonevalterm?

**Comment by prabirshrestha** [comment]
We can check this in once we have the event queue and diff changes in. Would help reduce talking too much to the server.

**Comment by prabirshrestha** [comment]
closing in favor of the new vim/neovim popup feature which also works in terminal and is more powerful than balloon. https://github.com/prabirshrestha/vim-lsp/pull/395

**Pull request closed without merging**